### PR TITLE
flux job list: make filtering options match flux jobs(1)

### PR DESCRIPTION
**Pull request by garlick**
This updates state filtering options in the `flux job list` plumbing command to match those in `flux-jobs(1)`, namely:
* Drop `--pending`, `--running`, and `--inactive` options
* Add `--states=STATES` where STATES a comma-separated list of states (virtual or real)
* Add `-A,--all-jobs` (all users, any job state)
* Add `-a,--all-my-jobs` (my jobs, any job state)

I went ahead and defined the `FLUX_JOB_PENDING`, `FLUX_JOB_RUNNING`, and `FLUX_JOB_ACTIVE` "virtual states" as a mask of actual RFC 21 states in `job.h`.  I structured `flux job list` to construct a state mask and derive the `job-info.list` request flags from that, to minimize required changes when #2628 (job-info: support listing jobs in specific states) lands.

Note that I had suggested that `t/job-manager/list-jobs` could be removed once this is done.  I misspoke - it actually needs to stay since it directly queries the job manager, since `job-info` is "eventually consistent".

### CODE REVIEW

**Comment by garlick** [comment]
Eeek, I choked the cffi generation script with my enum containing constant expressions.  Pushed a fixup.

**Comment by garlick** [comment]
Squashed the fixup to make the pr validator happy now that travis is passing.

**Comment by grondo** [comment]
Nice!

I hate to pick on the little things but `flux jobs -a, -A` don't have long options equivalents, should we keep the same in `flux job list`? If we want to add longopts equivalents to `flux jobs` I'm not a fan of `flux job list --all-jobs` as redundant (maybe just `flux job list --all` and for `-a, --all-user`?)

File: src/common/libjob/job.h [hunk 1]
```diff
@@ -51,6 +51,14 @@ typedef enum {
     FLUX_JOB_INACTIVE               = 32,   // captive end state
```
Comment (chu11) [inline src/common/libjob/job.h]: English in comment:

> Updating the flux-info.list protocol so that the list flags
can be dropped will be for another PR against issue #2628.

"will be for" -> "in"?

**Comment by chu11** [comment]
> I hate to pick on the little things but flux jobs -a, -A don't have long options equivalents, should we keep the same in flux job list? If we want to add longopts equivalents to flux jobs I'm not a fan of flux job list --all-jobs as redundant (maybe just flux job list --all and for -a, --all-user?)

I left long opts out specifically because there are no long opts equivalents for `ps(1)`.  And coming up with a long opts for `-a` and `-A` would ultimately lead to redundant sounding option names.

`--all` and `--all-user` doesn't sound so bad though.

**Comment by grondo** [comment]
>  left long opts out specifically because there are no long opts equivalents for ps(1). And coming up with a long opts for -a and -A would ultimately lead to redundant sounding option names.

I figured that is why you didn't include long-opts for `-a` `-A`

**Comment by garlick** [comment]
Right, I'll just drop the long opts in `flux job list`.

File: src/common/libjob/job.h (hunk 1 above)
Comment (garlick) [reply src/common/libjob/job.h]: Yeah, sorry awkward sentence, and probably something more appropriate to communicate in the PR rather than a commit.  I'll add #2628 to the mention in the earlier sentence.

**Comment by garlick** [comment]
Forced a push the the long option name changes
```
Usage: flux-job list [OPTIONS]
List jobs
  -A, --all              List jobs for all users, regardless of state
  -a, --all-user         List my jobs, regardless of state
  -c, --count=N          Limit output to N jobs
  -h, --help             Display this message.
  -s, --states=STATES    List jobs in specific states
  -u, --userid=UID       Limit output to specific userid. Specify "all" for all
                         users.
```
and commit message fix up.

**Comment by grondo** [comment]
Probably not a big deal but `--state=active` `--state=inactive` don't seem to be tested according to codecov. You could get a small coverage bump if we added tests for those flags.

**Comment by garlick** [comment]
The "inactive" leg of the that conditional could not be reached because it's a real state, and the "strtostate" leg was catching it, so deleted.

I converted one of the existing `flux job list -a` calls to `flux job list -s active,inactive` to cover the active state.

Forced a push - hopefully that will get coverage up a bit.

**Comment by garlick** [comment]
restarted a builder that failed with
> ERROR: t1001-kvs-internals.t - exited with status 137 (terminated by signal 9?)
